.github/workflows/generate-summary.yaml

```diff
@@ -187,11 +191,70 @@ jobs:
           unzip ./temp/*report.log.zip -d ./current_logs || true
           ls current_logs
 
+      - name: Build Log Comparison Setup
+        id: build-log-setup
+        run: |
+          echo "previous_build_logs_path=$(readlink -f previous_build_logs)" >> $GITHUB_OUTPUT
+          echo "previous_build_log_zips_path=$(readlink -f previous_build_log_zips)" >> $GITHUB_OUTPUT
+          echo "current_build_logs_path=$(readlink -f current_build_logs)" >> $GITHUB_OUTPUT
+          echo "current_build_log_zips_path=$(readlink -f current_build_log_zips)" >> $GITHUB_OUTPUT
+          mkdir new_build_warnings
+          echo "new_build_warnings_path=$(readlink -f ./new_build_warnings/new_build_warnings.log)" >> $GITHUB_OUTPUT
+
       - name: Download artifacts
+        id: download-previous-artifacts
         run: |
           pip install pygithub requests
-          python ./scripts/download_artifacts.py -hash ${{ inputs.gcchash }} -repo patrick-rivos/gcc-postcommit-ci -token ${{ secrets.GITHUB_TOKEN }} -prefix "${{ inputs.prefix }}"
+          python ./scripts/download_artifacts.py -hash ${{ inputs.gcchash }} -repo patrick-rivos/gcc-postcommit-ci -token ${{ secrets.GITHUB_TOKEN }} -prefix "${{ inputs.prefix }}" -build-logs -build-logs-dir ${{ steps.build-log-setup.outputs.previous_build_log_zips_path }}
           ls previous_logs
+          if [ -d "${{ steps.build-log-setup.outputs.previous_build_log_zips_path }}" ]; then
+            ls ${{ steps.build-log-setup.outputs.previous_build_log_zips_path }}
+          fi
+
+      - name: Unzip previous build log artifacts
+        id: unzip-previous-build-log-artifacts
+        uses: ./.github/actions/common/unzip-all-zips
+        with:
+          input-dir: ${{ steps.build-log-setup.outputs.previous_build_log_zips_path }}
+          output-dir: ${{ steps.build-log-setup.outputs.previous_build_logs_path }}
+          include-pattern: "*-stderr.log"
+
+      - name: Download current build log artifacts
+        id: download-current-build-log-artifacts
+        if: github.event_name != 'workflow_dispatch'
+        uses: actions/download-artifact@v4
+        with:
+          pattern: "*-build-log"
+          merge-multiple: true
+          path: ${{ steps.build-log-setup.outputs.current_build_log_zips_path }}
+
+      - name: Download current build log artifacts manual
+        id: download-current-build-log-artifacts-manual
+        if: (github.event_name == 'workflow_dispatch') && (inputs.build_log_current_run_id != '')
+        uses: actions/download-artifact@v4
+        with:
+          pattern: "*-build-log"
+          merge-multiple: true
+          path: ${{ steps.build-log-setup.outputs.current_build_log_zips_path }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          run-id: ${{ inputs.build_log_current_run_id }}
```

Comment:
Can we handle this similar to the testsuite logs so we don't need to specify the run id?

Reply:
Is manual workflow dispatch not only for testing purposes. If it's also used to view the generated summary of a arbitrary gcc hash, it seems to be better to follow the testsuite logs flow

Reply:
I use it to re-generate issues if something has gone wrong in the generate-summary step.
Currently I just specify the hash and it takes it from there - so also having to grab the run id isn't ideal.
Also if there are multiple baselines (due to build fails or whatever) that can't be specified with a single run_id but it's handled with a hash/the testsuite logs flow.

Reply:
Currently this run_id is only used to grab the current run, so there would be only one hash used for this purpose. Baseline hashes are handled using the testsuite logs flow already.

Reply:
I'd prefer only specifying the hash and automatically grabbing artifacts across workflows using the hash via the api.